azerothcore/azerothcore-wotlk
# fix(Scripts/Ulduar): Expedition Base Camp

## CHANGES PROPOSED:
Script the intro in the Ulduar's Base Camp

This consist of:
- Fixing the Gossip Menus for `Brann Bronzebeard` and `Lore Keeper of Norgannon`.
- Adding roleplay elements to both menus mentioned above.
- Script the `Transitus Shield Beam` spell that the `Kiring Tor Battle-Mages` cast
- Eliminate the script from cpp and add everything to SAI

## ISSUES ADDRESSED:
- Closes https://github.com/azerothcore/azerothcore-wotlk/pull/3548
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/3528
- Partially fixes: https://github.com/azerothcore/azerothcore-wotlk/issues/570



## TESTS PERFORMED:
- Tested in game
- Tested in 10 man and 20 man
- Tested on server restart
- Tested on Windows



## HOW TO TEST THE CHANGES:

`.tele Ulduar`
To be able to join the raid without being in a raid group:
`.gm on`
There are two ways to start the instance!!
- One starts from the `Lore Keeper of Norgannon`, to test the changes here, he has 2 gossip menus and one script per menu:
1. There will be a specific conversation between `High Explorer Dellorah` and `Lore Keeper of Norgannon` once the first gossip is pressed, please stay and look at it completely, you can even watch `Archmage Rhydian` moving and annoucing `Brann` about what the `Lore Keeper` informs us trough `Dellorah` ( she will have text emotes in chat ).
2. To start the encounter press the second gossip option that says you confirm you want to start the easier instance, watch the script now.
- One starts from the npc `Brann Bronzebeard`
1. For `Brann Bronzebeard` and the other npcs in this script, it is the same as the script done on confirming the option from the `Lore Keeper`, only that now the towers are not activated, the roleplay is still the same tho

The `Kirin Tor Battle-Mages` from the issue and pull:
https://github.com/azerothcore/azerothcore-wotlk/issues/570
https://github.com/azerothcore/azerothcore-wotlk/pull/3548

Are now fixed and the spell is correctly scripted.

Videos i used for this:
https://www.youtube.com/watch?v=m4fDOMYyjWw
https://www.youtube.com/watch?v=5bqZAvDY39U
https://www.youtube.com/watch?v=fQB4XhhHP6o
https://www.youtube.com/watch?v=VIB8pAXeG-0&t=17m20s
https://www.youtube.com/watch?v=FHseEYoFSy0
https://www.youtube.com/watch?v=xlT0idl-tMw


## KNOWN ISSUES AND TODO LIST:

- [X] If you start the roleplay from `Lore Keeper of Norgannon` ( the first one, not the instance script start ) and then you start the instance on easy mode from Brann, the script between them still continues and Archmage Rhydian is still sent to notify Brann Bronzebeard


## Target branch(es):








 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR
